server/controllers/report.js

```diff
@@ -0,0 +1,142 @@
+const db = require('../models');
+const { keyBy } = require('lodash');
+
+function findBySubject({ query: { id } }, res) {
+  return db.report.findAll({
+    where: { subjectId: id },
+    include: [
+      {
+        as: 'reporter',
+        model: db.user
+      }
+    ]
+  })
+    .then(reports => res.send(reports));
+}
+
+function getLastUsersReports({ user: { role, team } }, res) {
+  const query = {
+    attributes: [
+      db.sequelize.literal('DISTINCT ON("report"."subject_id") 1'),
+      'id', 'createdAt'
+    ],
+    order: [['subject_id', 'asc'], ['createdAt', 'desc']],
+    include: [{
+      as: 'subject',
+      model: db.user,
+      attributes: ['id', 'firstName', 'lastName', 'office'],
+      required: true
+    }]
+  };
+
+  if (role !== 'ADMIN') {
+    const [userJoin] = query.include;
+    userJoin.where = { team };
```

Comment:
Rework the way you do this because it is backwords

Reply:
Did you mean that I include where in query at start and then check if user is admin and then omit where from query?? 🤔

Reply:
You are building a query with an include and if something isn't a specific value you extract the first member of the include array and add something to it. To me that is convoluted and I strongly believe you can find a better way to do it